amazon_transcribe/serialize.py

```diff
@@ -25,62 +24,104 @@
     BaseEvent,
 )
 
-HEADER_VALUE = Union[int, None, str]
 
-
-class Serializer:
-    def __init__(self):
-        raise NotImplementedError("Serializer")
-
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        """Serialize out to payload and headers."""
-        raise NotImplementedError("serialize")
-
-    def serialize_to_request(self) -> PreparedRequest:
-        """Serialize parameters into an HTTP request."""
-        raise NotImplementedError("serialize_to_request")
-
-
-class TranscribeStreamingRequestSerializer(Serializer):
+class TranscribeStreamingSerializer:
     """Convert StartStreamTranscriptionRequest into a
     Request object for streaming to the Transcribe service.
     """
 
-    def __init__(self, endpoint, transcribe_request):
-        self.endpoint: str = endpoint
-        self.method: str = "POST"
-        self.request_uri: str = "/stream-transcription"
-        self.request_shape: StartStreamTranscriptionRequest = transcribe_request
+    def _serialize_str_header(self, header, value):
+        return {header: str(value)}
 
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        headers = {
-            "x-amzn-transcribe-language-code": self.request_shape.language_code,
-            "x-amzn-transcribe-sample-rate": self.request_shape.media_sample_rate_hz,
-            "x-amzn-transcribe-media-encoding": self.request_shape.media_encoding,
-            "x-amzn-transcribe-vocabulary-name": self.request_shape.vocabulary_name,
-            "x-amzn-transcribe-session-id": self.request_shape.session_id,
-            "x-amzn-transcribe-vocabulary-filter-method": self.request_shape.vocab_filter_method,
-            "x-amzn-transcribe-vocabulary-filter-name": self.request_shape.vocab_filter_name,
-            "x-amzn-transcribe-show-speaker-label": self.request_shape.show_speaker_label,
-            "x-amzn-transcribe-enable-channel-identification": self.request_shape.enable_channel_identification,
-            "x-amzn-transcribe-number-of-channels": self.request_shape.number_of_channels,
-        }
+    def _serialize_int_header(self, header, value):
```

Comment:
So I think the goal of having separate functions like this was to do some kind of high level validation and any datatype specific modification. You could also potentially handle `None` cases to reduce the amount of checking you're trying to do below before constructing the headers dictionary.

i.e.
```python
def _serialize_int_header(self, header, value):
    if not isinstance(value, int):
        raise ParamValidationError(f'{value} doesn't match expected type int')
    elif value is None:
         return {}
    else:
        return {header: str(value)}
```